src/opm/parser/eclipse/EclipseState/Runspec.cpp

```diff
@@ -457,6 +459,43 @@ bool SatFuncControls::operator==(const SatFuncControls& rhs) const
         && (this->family() == rhs.family());
 }
 
+Nupcol::Nupcol() :
+    min_nupcol(ParserKeywords::MINNPCOL::VALUE::defaultValue),
+    nupcol_value(ParserKeywords::NUPCOL::NUM_ITER::defaultValue)
+{}
+
+Nupcol::Nupcol(const Deck& deck) :
+    Nupcol()
+{
+    const RUNSPECSection runspecSection{deck};
+    if (runspecSection.hasKeyword<ParserKeywords::MINNPCOL>()) {
+        const auto& min_item = runspecSection.getKeyword<ParserKeywords::MINNPCOL>().getRecord(0).getItem<ParserKeywords::MINNPCOL::VALUE>();
+        this->min_nupcol = min_item.get<int>(0);
+    }
+}
+
+void Nupcol::update(int value) {
+    if (value < this->min_nupcol)
+        OpmLog::note(fmt::format("OPM Flow uses {} as minimum NUPCOL value", this->min_nupcol));
+    this->nupcol_value = std::max(value, this->min_nupcol);
```

Comment:
Is this a place where we should point the user to MINNPCOL keyword

Reply:
Good idea!

Reply:
I can make a PR with some additional notes to the user + changing the default.